demo/src/setup.ts

```diff
@@ -42,13 +46,84 @@ window.MonacoEnvironment = {
   }
 }
 
+Services.install({
+  workspace: {
+    workspaceFolders: [{
+      index: 0,
+      name: 'toto',
+      uri: monaco.Uri.file('/')
+    }]
+  }
+})
+
+let currentEditor: ({
+  modelRef: IReference<IResolvedTextEditorModel>
+  editor: monaco.editor.IStandaloneCodeEditor
+} & monaco.IDisposable) | null = null
+const openNewCodeEditor: OpenEditor = async (modelRef) => {
+  if (currentEditor != null) {
+    currentEditor.dispose()
+    currentEditor = null
+  }
+  const container = document.createElement('div')
+  container.style.position = 'fixed'
+  container.style.backgroundColor = 'rgba(0, 0, 0, 0.5)'
+  container.style.top = container.style.bottom = container.style.left = container.style.right = '0'
+  container.style.cursor = 'pointer'
+
+  const editorElem = document.createElement('div')
+  editorElem.style.position = 'absolute'
+  editorElem.style.top = editorElem.style.bottom = editorElem.style.left = editorElem.style.right = '0'
+  editorElem.style.margin = 'auto'
+  editorElem.style.width = '80%'
+  editorElem.style.height = '80%'
+
+  container.appendChild(editorElem)
+
+  document.body.appendChild(container)
+  try {
+    const editor = createConfiguredEditor(
+      editorElem,
+      {
+        model: modelRef.object.textEditorModel,
+        readOnly: true,
+        automaticLayout: true
+      }
+    )
+
+    currentEditor = {
+      dispose: () => {
+        editor.dispose()
+        modelRef.dispose()
+        document.body.removeChild(container)
+        currentEditor = null
+      },
+      modelRef,
+      editor
+    }
+
+    editor.onDidBlurEditorText(() => {
+      currentEditor?.dispose()
+    })
+    container.addEventListener('mousedown', (event) => {
+      if (event.target !== container) {
+        return
+      }
+
+      currentEditor?.dispose()
+    })
+
+    return editor
+  } catch (error) {
+    document.body.removeChild(container)
+    currentEditor = null
+    throw error
+  }
+}
+
 // Override services
 StandaloneServices.initialize({
-  ...getModelEditorServiceOverride(async (model, options) => {
-    // eslint-disable-next-line no-console
-    console.log('trying to open a model', model, options)
-    return undefined
-  }),
+  ...getModelEditorServiceOverride(openNewCodeEditor),
```

Comment:
Can you explain what the purpose of this `openNewCodeEditor` is`?

Reply:
It's a function that is called when a new editor should be opened to display a model given as parameter

It happens for instance when you ctrl+click on an import of another file with LSP enabled (and when there is a file content provider registered)

It can also happen with the debugger: try to enter the `console.log` call while debugging and you'll see another editor in a popup containing a node source code file